AdobeIms/Test/Unit/Model/UserProfileTest.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeIms\Test\Unit\Model;
+
+use Magento\AdobeIms\Model\UserProfile;
+use Magento\AdobeImsApi\Api\Data\UserProfileExtensionInterface;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * User profile test.
+ *
+ * Tests all setters and getters of data transport class
+ */
+class UserProfileTest extends TestCase
+{
+    /**
+     * @var ObjectManager
+     */
+    private $objectManager;
+
+    /**
+     * @var UserProfile $model
+     */
+    private $model;
+
+    /**
+     * Prepare test object.
+     */
+    protected function setUp(): void
+    {
+        $this->objectManager = new ObjectManager($this);
+        $this->model = $this->objectManager->getObject(UserProfile::class);
+    }
+
+    /**
+     * Test setAccessToken
+     *
+     * @param string|null $value
+     * @testWith ["value1"]
```

Comment:
This feature available only for PHPUnit 7.1 Magento uses PHPUnit 6.5.14 https://phpunit.readthedocs.io/en/7.1/annotations.html#testwith